src/features/decoration/decoration-placer.ts

```diff
@@ -0,0 +1,65 @@
+/********************************************************************************
+ * Copyright (c) 2017-2018 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import { injectable } from "inversify";
+import { VNode } from "snabbdom/vnode";
+import { SModelElement, SChildElement } from "../../base/model/smodel";
+import { IVNodeDecorator } from "../../base/views/vnode-decorators";
+import { isDecoration, Decoration } from "./model";
+import { setAttr } from "../../base/views/vnode-utils";
+import { Point, ORIGIN_POINT } from "../../utils/geometry";
+import { SEdge } from "../../graph/sgraph";
+import { isSizeable } from "../bounds/model";
+
+@injectable()
+export class DecorationPlacer implements IVNodeDecorator {
+    decorate(vnode: VNode, element: SModelElement): VNode {
+        if (isDecoration(element)) {
+            const position = this.getPosition(element);
+            const translate = 'translate(' + position.x + ', ' + position.y + ')';
+            setAttr(vnode, 'transform', translate);
+        }
+        return vnode;
+    }
+
+    protected getPosition(element: SModelElement & Decoration): Point {
+        if (element instanceof SChildElement && element.parent instanceof SEdge) {
```

Comment:
Generalize to `isRoutable(element.parent)`

Reply:
What else is routable if not an edge?

Reply:
It's more a dependency issue. We have to be careful about dependencies from src/graph to src/features and back.